### PR TITLE
Rebase to ocp/release:4.2.0-0.nightly-2019-07-12-041904

### PR DESCRIPTION
**Pull request by hardys**
Rebase on the latest openshift/installer now that some of our PRs have merged, and also to get an updated version of RHCOS that should unblock #100

### CODE REVIEW

File: data/data/baremetal/main.tf [hunk 1]
```diff
@@ -7,6 +7,11 @@ provider "ironic" {
   microversion = "1.52"
 }
 
+provider "ironic" {
+  url          = var.ironic_uri
+  microversion = "1.52"
+}
+
```
Comment (hardys) [inline data/data/baremetal/main.tf]: This is a rebase mistake

**Comment by hardys** [comment]
@stbenjam and @markmc this has been tested via https://github.com/openshift-metal3/dev-scripts/pull/672 and appears to be working, if you get time to review that would be much aprpeciated :)

File: data/data/baremetal/main.tf [hunk 2]
```diff
@@ -20,6 +20,7 @@ module "bootstrap" {
 module "masters" {
   source = "./masters"
 
+  ironic_uri     = var.ironic_uri
```
Comment (stbenjam) [inline data/data/baremetal/main.tf]: I think this shouldn't be here, we removed it in our repo but not from openshift/installer yet

File: data/data/baremetal/masters/variables.tf [hunk 3]
```diff
@@ -1,8 +1,14 @@
+variable "ironic_uri" {
+  type        = string
+  description = "URI for Ironic that bootstrap should wait for to be available"
+}
+
```
Comment (stbenjam) [inline data/data/baremetal/masters/variables.tf]: Same here

**Review by stbenjam — COMMENTED** [review]
Otherwise I think this looks good

File: data/data/baremetal/main.tf (hunk 2 above)
Comment (hardys) [reply data/data/baremetal/main.tf]: Ah Ok, will fix and re-test

Comment (hardys) [reply data/data/baremetal/main.tf]: Thanks, fixed

File: data/data/baremetal/masters/variables.tf (hunk 3 above)
Comment (hardys) [reply data/data/baremetal/masters/variables.tf]: Thanks, fixed

File: pkg/terraform/exec/plugins/Gopkg.lock [hunk 4]
```diff
@@ -795,6 +795,14 @@
   revision = "9f5bba54dad78c10ed572eb2541f8178a5500656"
   version = "v0.1.4"
 
+[[projects]]
+  digest = "1:a0886267c60262a245d6eeb0be12bf8e165026ad088bce106b9c55c5d613144f"
+  name = "github.com/openshift-metalkube/terraform-provider-ironic"
```
Comment (stbenjam) [inline pkg/terraform/exec/plugins/Gopkg.lock]: Ooh sorry looks like this got included too, we should only use the metal3 version, I'd remove this from Gopkg.lock and the openshift-metalkube directory in vendor

**Review by stbenjam — APPROVED** [review]
lgtm

**Comment by stbenjam** [comment]
CI passed going to merge this.

**Pull request merged**